kfs-core/src/main/java/edu/arizona/kfs/fp/document/DisbursementVoucherDocument.java

```diff
@@ -96,33 +99,29 @@ public void prepareForSave() {
             accountingLineExtension.setDocumentNumber(accountingLine.getDocumentNumber());
             accountingLineExtension.setSequenceNumber(accountingLine.getSequenceNumber());
         }
-        
-     // START KATTS-1961 Tag and JSP for DV, PREQ and CM Documents
+
         getIncomeTypeHandler().removeZeroValuedIncomeTypes();
-        // END KATTS-1961
-        
-        //KATTS-1939 Add New Search Fields to DV, PREQ and CM Documents 
-        // Only update paid year if the document is in final status 
-        if (KewApiConstants.ROUTE_HEADER_FINAL_CD.equals( getDocumentHeader().getWorkflowDocument().getStatus().getCode() ) ) {
-        	if (paidDate != null) {
-                setDvPaidYear(getPaidDate().toString().substring(0, 4));                
-            }
-            else {
-               setDvPaidYear(null);
+
+        // Only update paid year if the document is in final status
+        if (KewApiConstants.ROUTE_HEADER_FINAL_CD.equals(getDocumentHeader().getWorkflowDocument().getStatus().getCode())) {
+            if (paidDate != null) {
+                setDvPaidYear(getPaidDate().toString().substring(0, 4));
+            } else {
+                setDvPaidYear(null);
             }
-        }        
+        }
+
+        setDv1099Ind(false);
         for (DisbursementVoucherIncomeType incomeType : getIncomeTypes()) {
-            if ((StringUtils.isBlank(incomeType.getIncomeTypeCode()) || incomeType.getIncomeTypeCode().equals(KFSConstants.IncomeTypeConstants.INCOME_TYPE_NON_REPORTABLE_CODE))) {
-                setDv1099Ind(false);
-            }
-            else {
+            boolean isCodeExist = StringUtils.isNotBlank(incomeType.getIncomeTypeCode());
+            boolean isReportable = !incomeType.getIncomeTypeCode().equals(KFSConstants.IncomeTypeConstants.INCOME_TYPE_NON_REPORTABLE_CODE);
+            if (isCodeExist && isReportable) {
                 setDv1099Ind(true);
-                break;

```

Comment:
Not that it's a huge hit, but wouldn't setting 'true' once on this property do the trick, and testing the remainder of the list isn't necessary? In other words, isn't preferable to keep the break? (This is another nit that won't stop a merge since n is really small, but if we hit something necessary, let's snag this one too.) BTW, I really like how this block is refactored to be more intuitive of what the intent is.